tests/utils/test_dict_ops.py

```diff
@@ -2,12 +2,41 @@
 
 import pytest
 from pydantic import BaseModel
+from pytest_mock import MockerFixture
 
-from kpops.utils.dict_ops import generate_substitution, update_nested_pair
+from kpops.utils.dict_ops import (
+    generate_substitution,
+    update_nested,
+    update_nested_pair,
+)
 from kpops.utils.types import JsonType
 
 
 class TestDictOps:
+    def test_update_nested(self, mocker: MockerFixture):
+        list_dic = [{"k1": {"foo": 1}}, {"k1": {"bar": ""}}, {"k1": {"baz": "2"}}]
```

Comment:
looks great!
I would just rename the input to something like `dicts`, it doesn't really matter if it's a tuple or list